### ISSUE TITLE
indexing a `FrequencySeries` returns a new `FrequencySeries` without indexed frequencies

### ISSUE THREAD

**Issue by jrollins** [gwpy/gwpy#1680]
```python
In [3]: from gwpy.timeseries import TimeSeries

In [4]: ts = TimeSeries.fetch('H1:CAL-DARM_ERR_DBL_DQ', 1378540810, 1378540818)

In [6]: psd = ts.psd()

In [7]: psd.size
Out[7]: 16385

In [19]: inds = np.where(psd.frequencies.value > 8000)[0]

In [20]: a = psd[inds]

In [21]: a.size
Out[21]: 384

In [22]: len(a.value)
Out[22]: 384

In [23]: len(a.frequencies)
Out[23]: 16385
```

Note the length of the `frequencies` attribute is the size of the original series, not the newly indexed series.

**Comment by jrollins** [comment]
This problem applies to the `take()` method as well, which should presumably be doing the same thing.

**Comment by jrollins** [comment]
FYI This issue appears to be fixed in 3.0.8.  Thank you all for addressing it.
